src/adapter/src/catalog/storage.rs

```diff
@@ -1898,14 +1955,16 @@ pub struct ItemValue {
     definition: SerializedCatalogItem,
 }
 
-#[derive(Clone, Deserialize, Serialize, PartialOrd, PartialEq, Eq, Ord, Hash)]
+#[derive(Clone, Deserialize, Serialize, PartialOrd, PartialEq, Eq, Ord, Hash, Debug)]
 pub struct RoleKey {
     id: RoleId,
 }
 
-#[derive(Clone, Deserialize, Serialize, PartialOrd, PartialEq, Eq, Ord)]
+#[derive(Clone, Deserialize, Serialize, PartialOrd, PartialEq, Eq, Ord, Debug)]
 pub struct RoleValue {
-    name: String,
+    // flatten needed for backwards compatibility.
+    #[serde(flatten)]
```

Comment:
I don't understand this. What does it do?

Reply:
It essentially inline's the struct definition for serialization and deserialization. So instead of looking like this on disk:
```
{
  role: {
    name: <name>,
    attributes: { .. },
  }
}
```
It will look like this this:
```
{
  name: <name>,
  attributes: { .. },
}
```

The reason we need it, is because in all current deployments, the roles are serialized as:
```
{
  name: <name>
}
```

So we need something with `name` in the top level to deserialize old roles.